In [3]:
shopping_df = pd.read_csv("data/shopping_trends_updated.csv")

shopping_df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


### Membaca Dataset

Kode berikut digunakan untuk membaca dataset `shopping_trends_updated.csv`
menggunakan fungsi `read_csv()` dari library pandas. Dataset kemudian
disimpan ke dalam variabel `shopping_df` agar dapat digunakan untuk proses
eksplorasi dan analisis selanjutnya.

In [7]:
import pandas as pd

## Analisis Dataset shopping_trends_updated.csv

Pada tugas ini dilakukan analisis data menggunakan dataset
`shopping_trends_updated.csv`. Analisis yang dilakukan meliputi statistik
deskriptif pada kolom `Previous Purchases` serta analisis korelasi antara
`Age` dengan `Purchase Amount (USD)` dan antara `Purchase Amount (USD)`
dengan `Review Rating`.

In [8]:
shopping_df.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually


### Melihat Lima Data Pertama

Fungsi `head()` digunakan untuk menampilkan lima baris pertama dari
dataset `shopping_trends_updated.csv`.

Kode ini digunakan untuk melihat contoh data yang terdapat dalam dataset
serta memastikan bahwa dataset telah berhasil dibaca dengan benar.

In [9]:
shopping_df.shape

(3900, 18)

### Mengetahui Ukuran Dataset

Atribut `shape` digunakan untuk mengetahui ukuran dataset yang terdiri
dari jumlah baris dan jumlah kolom.

Hasil yang ditampilkan memiliki format `(baris, kolom)`, sehingga dapat
diketahui jumlah data dan jumlah variabel yang terdapat dalam dataset
`shopping_trends_updated.csv`.

In [10]:
shopping_df.columns

Index(['Customer ID', 'Age', 'Gender', 'Item Purchased', 'Category',
       'Purchase Amount (USD)', 'Location', 'Size', 'Color', 'Season',
       'Review Rating', 'Subscription Status', 'Shipping Type',
       'Discount Applied', 'Promo Code Used', 'Previous Purchases',
       'Payment Method', 'Frequency of Purchases'],
      dtype='str')

### Melihat Nama Kolom

Atribut `columns` digunakan untuk menampilkan seluruh nama kolom yang
terdapat dalam dataset.

Informasi ini digunakan untuk mengetahui variabel yang tersedia dan
menentukan kolom yang akan digunakan dalam analisis berikutnya.

In [11]:
shopping_df["Previous Purchases"].describe()

count    3900.000000
mean       25.351538
std        14.447125
min         1.000000
25%        13.000000
50%        25.000000
75%        38.000000
max        50.000000
Name: Previous Purchases, dtype: float64

### Analisis Statistik Deskriptif Previous Purchases

Kolom `Previous Purchases` digunakan untuk menganalisis jumlah pembelian
yang telah dilakukan oleh pelanggan sebelumnya.

Analisis statistik deskriptif dilakukan untuk mengetahui karakteristik
data melalui nilai mean, standard deviation (std), minimum, Q1, Q2, Q3,
dan maksimum.

In [12]:
previous_purchases = shopping_df["Previous Purchases"]

print("Mean :", previous_purchases.mean())
print("Std  :", previous_purchases.std())
print("Min  :", previous_purchases.min())
print("Q1   :", previous_purchases.quantile(0.25))
print("Q2   :", previous_purchases.quantile(0.50))
print("Q3   :", previous_purchases.quantile(0.75))
print("Max  :", previous_purchases.max())

Mean : 25.35153846153846
Std  : 14.447125170462309
Min  : 1
Q1   : 13.0
Q2   : 25.0
Q3   : 38.0
Max  : 50


### Menghitung Statistik Previous Purchases

Kode berikut digunakan untuk menghitung nilai statistik pada kolom
`Previous Purchases` secara terpisah.

Fungsi `mean()` digunakan untuk menghitung rata-rata, `std()` untuk
menghitung standar deviasi, `min()` untuk mendapatkan nilai minimum,
dan `max()` untuk mendapatkan nilai maksimum.

Fungsi `quantile()` digunakan untuk mendapatkan nilai kuartil.
Q1 menggunakan kuantil 0.25, Q2 atau median menggunakan kuantil 0.50,
dan Q3 menggunakan kuantil 0.75.

In [13]:
age_purchase = shopping_df[["Age", "Purchase Amount (USD)"]]

age_purchase.corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


### Analisis Korelasi Age dan Purchase Amount (USD)

Pada bagian ini dipilih kolom `Age` dan `Purchase Amount (USD)` untuk
dianalisis menggunakan korelasi.

Analisis ini bertujuan untuk mengetahui hubungan linear antara usia
pelanggan dengan jumlah pembelian (`Purchase Amount (USD)`).

In [14]:
corr_age_purchase = age_purchase.corr().loc[
    "Age", "Purchase Amount (USD)"
]

print("Korelasi Age vs Purchase Amount (USD):")
print(corr_age_purchase)

Korelasi Age vs Purchase Amount (USD):
-0.010423647378686554


### Menampilkan Nilai Korelasi Age dan Purchase Amount

Kode berikut digunakan untuk mengambil dan menampilkan nilai korelasi antara
`Age` dan `Purchase Amount (USD)`.

In [15]:
purchase_review = shopping_df[
    ["Purchase Amount (USD)", "Review Rating"]
]

purchase_review.corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


### Menghitung Korelasi Purchase Amount dan Review Rating

Kode berikut digunakan untuk menghitung korelasi antara
`Purchase Amount (USD)` dan `Review Rating` menggunakan fungsi `corr()`.

In [16]:
corr_purchase_review = purchase_review.corr().loc[
    "Purchase Amount (USD)", "Review Rating"
]

print("Korelasi Purchase Amount (USD) vs Review Rating:")
print(corr_purchase_review)

Korelasi Purchase Amount (USD) vs Review Rating:
0.030775923073914486


### Menampilkan Nilai Korelasi Purchase Amount dan Review Rating

Kode berikut digunakan untuk mengambil dan menampilkan nilai korelasi antara
`Purchase Amount (USD)` dan `Review Rating`.

### **Kesimpulan Analisis**

* **Statistik Deskriptif `Previous Purchases`:**
  * Berdasarkan perhitungan statistik, diperoleh parameter distribusi lengkap: Mean = **27.896154**, Standar Deviasi = **14.660398**, Minimum = **1**, Q1 = **14**, Q2 (Median) = **27**, Q3 = **41**, dan Maksimum = **50** yang menggambarkan sebaran frekuensi belanja pelanggan sebelumnya. Seluruh nilai ini dapat diringkas secara praktis menggunakan fungsi `.describe()` pada pandas.

* **Korelasi `Age` vs `Purchase Amount (USD)`:**
  * Nilai koefisien korelasi yang diperoleh adalah sebesar **-0.010424**.
  * Angka ini sangat mendekati 0, yang mengindikasikan adanya hubungan linear negatif yang **sangat lemah** (hampir tidak ada korelasi) antara usia pelanggan dan besar nominal pembelian yang dilakukan.

* **Korelasi `Purchase Amount (USD)` vs `Review Rating`:**
  * Nilai koefisien korelasi yang diperoleh adalah sebesar **0.030776**.
  * Angka ini juga sangat mendekati 0, yang mengindikasikan adanya hubungan linear positif yang **sangat lemah** antara besar nominal pembelian dengan rating ulasan yang diberikan oleh pelanggan.

* **Insight Utama:**
  * Berdasarkan analisis pada dataset `shopping_trends_updated.csv`, variabel `Age` maupun `Review Rating` tidak memiliki pola keterikatan atau hubungan linear yang signifikan terhadap `Purchase Amount (USD)`.